apps/examples/memory_fragmentation_test/memory_fragmentation_test.c

```diff
@@ -0,0 +1,344 @@
+	int numof_size[MAX_SIZE_EXPONENT] = {700, 300, 200, 0, 0, 0, 0, 0, 70, 50, 0, 0}; 
+
+	for (i = 0; i < MAX_SIZE_EXPONENT; ++i) {
+		list[i].data = NULL;
+		list[i].next = NULL;
+		list[i].prev = NULL;
+	}
+
+	if (argc < 4) {
+		print_usage();
+	}
+
+	f1 = strtol(argv[2], (char **)NULL, 10);
+	if ((f1 < 1) || (f1 > 9)) {
+		print_usage();
+	}
+
+	f2 = strtol(argv[3], (char **)NULL, 10);
+	if ((f2 < 1) || (f2 > 9)) {
+		print_usage();
+	}
+
+	r = strtol(argv[4], (char **)NULL, 10);
+	if (r < 0) {
+		print_usage();
+	}
+
+	srand(SEED);
+
+	/* Allocate memory according to 'numof_size' */
+	if (memory_allocation(list, numof_size, num_alloc) == false) {
+		printf("memory_allocation failed!\n");
+		memory_cleanup(num_alloc, list);
+		return;
+	}
+
+	/* Free the number of memory allocation according to 'num_free' 
+	 * with a first-come first-free manner.
+	 */
+	printf("Test Configuration:\n");
+	printf("Size [bytes]	# of init alloc		# of free\n");
+	for (i = 0; i < NUMOF_SMALL_MEMORY; ++i) {
+		num_free[i] = num_alloc[i] * f1 / 10;
+		if (num_alloc[i] > 0) {
+			printf("%d		%d			%d\n", 1 << (i + 4), num_alloc[i], num_free[i]);
+		}
+	}
+	for (i = NUMOF_SMALL_MEMORY; i < MAX_SIZE_EXPONENT; ++i) {
+		num_free[i] = num_alloc[i] * f2 / 10;
+		if (num_alloc[i] > 0) {
+			printf("%d		%d			%d\n", 1 << (i + 4), num_alloc[i], num_free[i]);
+		}
+	}
+	if (memory_free(num_free, num_alloc, list) == false) {
+		printf("memory_free failed!\n");
+		memory_cleanup(num_alloc, list);
+		return;
+	}
+
+	/* Repeat a cycle of allocation and free a given number of times */
+	for (i = 0; i < r; ++i) {
+		/* memory allocation according to 'num_free' */
+		for (j = 0; j < MAX_SIZE_EXPONENT; ++j) {
+			num_alloc_tmp[j] = 0;
+		}
+		if (memory_allocation(list, num_free, num_alloc_tmp) == false) {
+			printf("memory_allocation failed!\n");
+			memory_cleanup(num_alloc, list);
+			return;
+		}
+		for (j = 0; j < MAX_SIZE_EXPONENT; ++j) {
+			num_alloc[j] += num_alloc_tmp[j];
+		}
+		
+		/* free allocation according to 'num_free' */
+		if (memory_free(num_alloc_tmp, num_alloc, list) == false) {
+			printf("memory_free failed!\n");
+			memory_cleanup(num_alloc, list);
+			return;
+		}
+	}
+
+	printf("\nMemory allocation & free repeated at %d times.\n", r);
+
+	/* wait for 10 seconds before free-ing all the memory */
+	printf("\nAll memory test has been done.\n");
+	printf("\nMemory allocation status: \n");
+	printf("Size [bytes]	# of alloc/free(repeat)	# of remaining alloc\n");
+	for (i = 0; i < MAX_SIZE_EXPONENT; ++i) {
+		if (num_alloc[i] > 0) {
+			printf("%d		%d			%d\n", 1 << (i + 4), num_free[i], num_alloc[i]);
+		}
+	}
+	printf("\nPlease, use 'heapinfo' to see how the heap memory is fragmented in detail.\n");
+}
+
+static int test_task(int argc, char *argv[])
+{
+	memory_fragmentation_test(argc, argv);
+
+	task_delete(0);
+	return 0;
+}
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int memfrag_main(int argc, char *argv[])
```

Comment:
I don't know why you make test_task and memory_fragmentation_test function seperately.
I think you can rename memory_fragmentation_test to memfrag_main and can delete this and test_task.

Reply:
Thanks for your comments. It's deleted.